### PR TITLE
feat: Board grouping

### PR DESCRIPTION
**Pull request by shawnlewis**
Replace the grouping control on the board with the new GroupingEditor.

The new control follows the style of the FilterEditor, it can be switched between visual/expression modes. Currently it just shows available string keys as grouping options, but we can make it help with other interesting grouping styles in the future.

image

### CODE REVIEW

File: weave-js/src/components/Panel2/PanelGroupingEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  listItemType: Type
+) => {
+  return opPick({
+    obj: varNode(listItemType, 'row'),
+    key: constString(groupingState.key),
+  });
+};
+
+const setGroupingKey = (
+  groupingState: VisualGroupingState,
```
Comment (jwlee64) [inline weave-js/src/components/Panel2/PanelGroupingEditor.tsx:94]: do we need to pass in groupingState here

**Review by jwlee64 — APPROVED** [review]
Manually tested, lgtm

File: weave-js/src/components/Panel2/PanelGroupingEditor.tsx (hunk 1 above)
Comment (shawnlewis) [reply weave-js/src/components/Panel2/PanelGroupingEditor.tsx:94]: No, but just following a pattern where we add these "action" style functions that modify a groupingState, returning a new one. When groupingState has more stuff in it, it'll make more sense :). But the FilterEditor does and you can see the pattern there.

Comment (jwlee64) [reply weave-js/src/components/Panel2/PanelGroupingEditor.tsx:94]: makes sense, the way it was set up i figured we might need it later

**Pull request merged**